streams/src/test/java/org/apache/kafka/streams/integration/StandbyTaskEOSIntegrationTest.java

```diff
@@ -329,26 +331,32 @@ private KafkaStreams buildWithDeduplicationTopology(final String stateDirPath) {
             Serdes.Integer())
         );
         builder.<Integer, Integer>stream(inputTopic)
-            .transform(
-                () -> new org.apache.kafka.streams.kstream.Transformer<Integer, Integer, KeyValue<Integer, Integer>>() {
+            .process(
+                () -> new Processor<Integer, Integer, Integer, Integer>() {
+                    private ProcessorContext<Integer, Integer> context;
                     private KeyValueStore<Integer, Integer> store;
 
-                    @SuppressWarnings("unchecked")
                     @Override
-                    public void init(final ProcessorContext context) {
-                        store = (KeyValueStore<Integer, Integer>) context.getStateStore(storeName);
+                    public void init(final ProcessorContext<Integer, Integer> context) {
+                        this.context = context;
+                        store = context.getStateStore(storeName);
                     }
 
                     @Override
-                    public KeyValue<Integer, Integer> transform(final Integer key, final Integer value) {
+                    public void process(final Record<Integer, Integer> record) {
+                        final Integer key = record.key();
+                        final Integer value = record.value();
+
                         if (skipRecord.get()) {
                             // we only forward so we can verify the skipping by reading the output topic
                             // the goal is skipping is to not modify the state store
-                            return KeyValue.pair(key, value);
+                            context.forward(record);
+                            return;
                         }
 
                         if (store.get(key) != null) {
-                            return null;
+                            store.delete(key);
```

Comment:
Why do we insert a delete here?

Reply:
I have thought it would be the most similar replacement for return null.